### PR TITLE
Onder lesson32 js

### PR DESCRIPTION
**Pull request by ondermartin**

### CODE REVIEW

**Review by annasvst — CHANGES_REQUESTED** [review]
Hi Önder, this is a very good start! Now RecipeDetail component is missing in your code. I see that there's a file in components folder created for it but it's empty. You need to add code to present recipe details. Some elements like div, heading, a couple of paragraphs and image is enough to display recipe data. Also, add back button in the Recipe details component and pass the 'handleClearSelection' to it to return to the main page.

File: JS/Lesson32/onder/my-app/src/App.jsx [hunk 1]
```diff
@@ -4,21 +4,29 @@ import {BD_SEARCH_BASE_URL} from './urls.js';
 import {RecipeList} from './components/RecipeList';
 import {SearchBar} from './components/SearchBar';
```
Comment (annasvst) [inline JS/Lesson32/onder/my-app/src/App.jsx:5]: This import is missing:
import { RecipeDetail } from './components/RecipeDetail';

File: JS/Lesson32/onder/my-app/src/components/RecipeDetail/index.jsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
+const RecipeDetail = ({ recipe, onClearSelection }) => {
```
Comment (annasvst) [inline JS/Lesson32/onder/my-app/src/components/RecipeDetail/index.jsx:3]: Must have: export the RecipeDetail component
```suggestion
export const RecipeDetail = ({ recipe, onClearSelection }) => {
```

File: JS/Lesson32/onder/my-app/src/components/RecipeDetail/index.jsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  return (
+    <div>
+      <h2>Recipe Details</h2>
+      <p>{recipe.strMeal}</p>
```
Comment (annasvst) [inline JS/Lesson32/onder/my-app/src/components/RecipeDetail/index.jsx:7]: Nice to have: add more details about the recipe:

```
  <div class="recipe-details-container">
      <h2 class="heading">Recipe Details</h2>
      <div class="image-content">
        <img class="recipe-thumbnail" src={recipe.strMealThumb} alt={recipe.strMeal} />
      </div>
      <div class="text-content">
        <p>{recipe.strMeal}</p>
        <p>{recipe.strInstructions}</p>
        <button onClick={onClearSelection}>Go Back to Recipes</button>
      </div>
```

File: JS/Lesson32/onder/my-app/src/components/RecipeDetail/index.jsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
```
Comment (annasvst) [inline JS/Lesson32/onder/my-app/src/components/RecipeDetail/index.jsx:2]: NIce to have: add styles to style the page nicely:

```suggestion
import './styles.css';
```

CSS code:
```
.recipe-details-container {
  display: grid;
  grid-template-rows: 50px 1fr;
  grid-template-columns: 1fr 1fr;
  grid-template-areas: "heading heading"
  "image-content text-content";
  gap: 20px;
}

.heading {
  grid-area: heading;
}

.image-content {
  grid-area: image-content;
}

.text-content {
  grid-area: text-content;
}

.recipe-thumbnail {
  width: 100%;
}
```

**Review by annasvst — CHANGES_REQUESTED** [review]
Hi Onur, you have code in App.js before that would set selected recipe when the user clicks on the recipe from the list. The code was removed in this commit: https://github.com/frontend-from-0/240123-exercises/pull/126/commits/2093d507b011fa30dcd2e0e359a3f0bd399b4706

Please re-add the missing part from the mentioned commit and re-submit

**Comment by ondermartin** [comment]
Hello Anna 
I have messed it up so much but hopefully this will fixed it I got so confused. HAHAHAHA

**Review by annasvst — APPROVED** [review]
Works now, approved!

**Pull request merged**